internal/keyring/keyring.go

```diff
@@ -0,0 +1,78 @@
+// Package keyring is a simple wrapper that adds timeouts to the zalando/go-keyring package.
+package keyring
+
+import (
+	"context"
+	"time"
+
+	"github.com/zalando/go-keyring"
+)
+
+type TimeoutError struct {
+	message string
+}
+
+func (e *TimeoutError) Error() string {
+	return e.message
+}
+
+// Set secret in keyring for user.
+func Set(service, user, secret string) error {
```

Comment:
Another option here that might be a minor improvement would be to get rid of the context in favour of a `time.After` i.e.

```go
func Set(service, user, secret string) error {
   ch := make(chan error, 1)
   go func() {
       defer close(ch)
       ch <- keyring.Set(service, user, secret)
    }()

    select {
    case err := <- ch:
        return err
    case <- time.After(time.Second)
        return &TimeoutError{"timeout while trying to set secret in keyring"}
    }
```

The reason I say minor improvement is that it's a touch more explicit. When I see a `context` I generally expect it to have come from somewhere and to be going somewhere. It also doesn't come with a footgun of not calling `cancel`.

Reply:
TIL `time.After`. Great suggestion. Implemented.